### ISSUE TITLE
Use Terraform-docs to generate module-description

### ISSUE THREAD

**Issue by strowi** [kube-hetzner/terraform-hcloud-kube-hetzner#203]
Hi,

not sure if you've come across [terraform-docs](https://terraform-docs.io/), but it produces a nice (and complete) module-description from code including all requirements, providers, modules, resources and inputs/outputs. 
Might be useful to incorporate this into the pipeline (or use it locally..).

**Comment by mysticaltech** [comment]
That would be great, @strowi; definitely something we need to do! 

If I remember well, @batthebee mentioned it somewhere, but I had forgotten about it! So thanks for the reminder.

**Comment by mysticaltech** [comment]
Solved in #205! 🥳

Just FYI @batthebee, it was still regenerating the output of the gh action in README.md, until I specified  `output-file: doc/terraform.md` also in the gh action workflow file (the terraform docs config was not enough, see below for reference), now all good!

Thanks again for this awesome addition @batthebee! 🚀  And initiative @strowi! 💙

```
::debug terraform-docs markdown table --config .terraform-docs.yaml --output-mode inject --output-file README.md --output-template <!-- BEGIN_TF_DOCS -->
{{ .Content }}
<!-- END_TF_DOCS --> .
README.md updated successfully
::debug No change in ./ detected
::debug Following files will be committed
M  README.md
[master 322b43e] terraform-docs: automated action
 1 file changed, 108 insertions(+), 1 deletion(-)
To https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner
   241fd12..322b43e  master -> master
```
